### PR TITLE
Allowing formatting non-copyable ranges.

### LINKED ISSUE #3286: Can't format move-only ranges
[Example](https://godbolt.org/z/cE1P9YdG6):

```cpp
#include <concepts>
#include <vector>
#include <fmt/ranges.h>

template <bool Copyable>
struct Vector {
    std::vector<int> v;

    Vector(std::initializer_list<int> elems) : v(elems) { }

    Vector(Vector&&) = default;
    Vector& operator=(Vector&&) = default;

    Vector(Vector const&) requires Copyable = default;
    Vector& operator=(Vector const&) requires Copyable = default;

    auto begin() { return v.begin(); }
    auto end() { return v.end(); }
};

static_assert(std::movable<Vector<false>>);
static_assert(std::movable<Vector<true>>);
static_assert(!std::copyable<Vector<false>>);
static_assert(std::copyable<Vector<true>>);

int main() {
    fmt::print("{}\n", Vector<true>{1, 2, 3});  // ok [1, 2, 3]
    fmt::print("{}\n", Vector<false>{1, 2, 3}); // error
}
```

This is because the range check for non-const ranges requires copyability and should probably just be removed (h/t @timsong-cpp):

https://github.com/fmtlib/fmt/blob/a2c05a10ec875712bfee739a05ed99fc3915f4e9/include/fmt/ranges.h#L154-L159

### PR DESCRIPTION
**Pull request by brevzin**
Fixes #3286

### CODE REVIEW

**Comment by brevzin** [comment]
Some windows build is complaining about `noncopyable_range<int> &` being non-formattable, but I'm not sure why. Do you know what version "(windows-2019, x64, v140, 14)" corresponds to on compiler-explorer? Could try to repro something there.

**Comment by mwinterb** [comment]
It's not an exact match, but the general "era" matches "x64 msvc v19.0 (WINE)".

**Comment by brevzin** [comment]
wat: https://godbolt.org/z/b9TYxqh6s

```cpp
namespace my {
namespace detail {
    template <typename... > using void_t = void;
}
}

template <typename T>
struct has_mutable_begin_end<
    T, 
    #if 1
    fmt::detail::void_t<
    #else
    my::detail::void_t<
    #endif
        decltype(fmt::detail::range_begin(std::declval<T>()))
        , decltype(fmt::detail::range_end(std::declval<T>()))
        #if 0
        , int
        #endif
        >>
    : std::true_type {};

static_assert(has_mutable_begin_end<noncopyable_range>::value, "");
```

If I use `my::detail::void_t`, this works. 

If I use `fmt::detail::void_t`, which is ... the same thing as `my::detail::void_t`, it fails. Unless I add the extra `int` at the end, in which case it works?

File: include/fmt/ranges.h [hunk 1]
```diff
@@ -155,7 +155,7 @@ template <typename T>
 struct has_mutable_begin_end<
     T, void_t<decltype(detail::range_begin(std::declval<T>())),
               decltype(detail::range_end(std::declval<T>())),
-              enable_if_t<std::is_copy_constructible<T>::value>>>
+              int>>
```
Comment (vitaut) [inline include/fmt/ranges.h:158]: Please add a comment briefly explaining why we have int here.

**Review by vitaut — COMMENTED** [review]
Thanks for the PR.

File: include/fmt/ranges.h (hunk 1 above)
Comment (brevzin) [reply include/fmt/ranges.h:160]: Sure.

**Pull request merged**